graph-builder/src/metadata.rs

```diff
@@ -0,0 +1,107 @@
+//! This module implements the fetching of dynamic metadata
+
+use failure::{Error, Fallible};
+use futures::future::Future;
+use quay;
+use registry;
+use std::path::PathBuf;
+
+pub static DEFAULT_QUAY_LABEL_FILTER: &str = "com.openshift.upgrades.graph";
+pub static MANIFESTREF_KEY: &str = "com.openshift.upgrades.graph.release.manifestref";
+
+pub struct MetadataFetchArgs {
```

Comment:
This is effectively just a string. Do you foresee some different usecase for the future? If not, I think it makes sense to switch to a plain `String`.